core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -323,6 +331,26 @@ private SaltConfig createSaltConfig(Stack stack, Cluster cluster, GatewayConfig
         return new SaltConfig(servicePillar, grainPropertiesService.createGrainProperties(gatewayConfigs, cluster, nodes));
     }
 
+    @VisibleForTesting
+    void addHostAttributes(Stack stack, Map<String, SaltPillarProperties> servicePillar, Set<Node> nodes) {
```

Comment:
I think this should be moved to a separate class together with `getAttributesForHostGroup`.
And this method should return with a map, like:
`Map.of("hostattrs", new SaltPillarProperties("/nodes/hostattrs.sls", singletonMap("hostattrs", attributes)))`
And in line 303 you would need only:
`servicePillar.putAll(createHostAttributes(..));`

I know there is a lot of code here (even written by me) where we add stuff to this map by passing it around but honestly it's not the best solution.

Moving this method out would make `@VisibleForTesting` unnecessary which is already a sign that it should be in a separate class.

Reply:
Moving this into a HostAttributeDecorator, and having that set the PillarProperties instead of returning a map.
I'm going to ask you to make changes to return a map if you feel strongly about that / other code style changes. Given I don't write a lot of CB patches - was avoiding any kind of re-structuring and following existing patterns as much as possible.